homeassistant/components/water_heater/__init__.py

```diff
@@ -39,12 +39,17 @@
 SUPPORT_TARGET_TEMPERATURE = 1
 SUPPORT_OPERATION_MODE = 2
 SUPPORT_AWAY_MODE = 4
+SUPPORT_TARGET_TEMPERATURE_HIGH = 8
+SUPPORT_TARGET_TEMPERATURE_LOW = 16
 
 ATTR_MAX_TEMP = 'max_temp'
 ATTR_MIN_TEMP = 'min_temp'
 ATTR_AWAY_MODE = 'away_mode'
 ATTR_OPERATION_MODE = 'operation_mode'
 ATTR_OPERATION_LIST = 'operation_list'
+ATTR_TARGET_TEMP_HIGH = 'target_temp_high'
```

Comment:
We shouldn't do this. If a water heater is set to auto, we should just assume it supports a high/low temp. 

We should not copy the mistakes from the climate component.

Reply:
Unsure what you mean here. Maybe comment is on wrong line. You mean the support features? the attributes are needed to be able to display. But the SUPPORT_TARGET_* are probably pointless. If the AUTO mode have no target min/max, they should just return None.

Reply:
@balloob can you clarify?

Reply:
oh right, this should be on the support ones.